pkg/execution/state/redis_state/lua/queue/backlogRequeue.lua

```diff
@@ -36,17 +37,8 @@ end
 if tonumber(redis.call("ZCARD", keyBacklogSet)) == 0 then
   redis.call("HDEL", keyBacklogMeta, backlogID)
 
-  redis.call("ZREM", keyShadowPartitionSet, backlogID)
-
-  -- If shadow partition has no more backlogs, update global/account pointers
-  if tonumber(redis.call("ZCARD", keyShadowPartitionSet)) == 0 then
-    redis.call("ZREM", keyGlobalShadowPartitionSet, partitionID)
-    redis.call("ZREM", keyAccountShadowPartitionSet, partitionID)
-
-    if tonumber(redis.call("ZCARD", keyAccountShadowPartitionSet)) == 0 then
-      redis.call("ZREM", keyGlobalAccountShadowPartitionSet, accountID)
-    end
-  end
+  -- Update pointers
+  updateBacklogPointer(keyShadowPartitionMeta, keyBacklogMeta, keyGlobalShadowPartitionSet, keyGlobalAccountShadowPartitionSet, keyAccountShadowPartitionSet, keyShadowPartitionSet, keyBacklogSet, accountID, partitionID, backlogID)
```

Comment:
This change is equivalent, except that the reusable function also clears out the metadata now.